Add pagination sorting swagger

petro2061 opened the pull request:
Added Pageable object to findAll method in Boot Controller class. Methods in the BookController class are also described, using the Open Api annotation.

petro2061 committed:
added openapi starter

petro2061 committed:
added Pageable object

petro2061 committed:
added open api configuration

petro2061 committed:
added configuration for open api

petro2061 committed:
Merge branch 'refs/heads/addValidationAndGHE'

petro2061 committed:
Merge branch 'main' of github.com:petro2061/online-book-store-project

petro2061 committed:
Merge branch 'main' into addPaginationSortingSwagger

petro2061 committed:
Merge branch 'refs/heads/main' into addPaginationSortingSwagger

# Conflicts:
#	pom.xml

petro2061 committed:
fix problem after merge into main

Elena-Bruyako commented on src/main/java/project/onlinebookstore/config/OpenApiConfig.java:
```diff
@@ -0,0 +1,19 @@
+
+@Configuration
+public class OpenApiConfig {
+
```
```suggestion
```

Elena-Bruyako commented on src/main/java/project/onlinebookstore/controller/BookController.java:
```diff
@@ -18,41 +22,71 @@
+    public List<BookDto> getAll(
+            @Parameter(description = "Standard object pageable "
+                    + "include page, size and sort")
+            Pageable pageable) {
```
```suggestion
            @ParameterObject @PageableDefault Pageable pageable) {
```

Elena-Bruyako commented on src/main/java/project/onlinebookstore/controller/BookController.java:
```diff
@@ -18,41 +22,71 @@
-    public BookDto getBookById(@PathVariable Long id) {
+    public BookDto getBookById(
+            @Parameter(description = "Represents the book identifier")
+            @PathVariable Long id) {
```
```suggestion
            @PathVariable @Positive Long id) {
```

Elena-Bruyako commented on src/main/java/project/onlinebookstore/controller/BookController.java:
```diff
@@ -18,41 +22,71 @@
-    public void delete(@PathVariable Long id) {
+    public void delete(
+            @Parameter(description = "Represents the book identifier")
+            @PathVariable Long id) {
```
```suggestion
            @PathVariable @Positive Long id) {
```

Elena-Bruyako commented on src/main/java/project/onlinebookstore/controller/BookController.java:
```diff
@@ -18,41 +22,71 @@
-                              @RequestBody @Valid CreateBookRequestDto updateBookDto) {
+    public BookDto updateBookById(
+            @Parameter(description = "Represents the book identifier")
+            @PathVariable Long id,
```
```suggestion
            @PathVariable @Positive Long id,
```

Elena-Bruyako replied:
not fixed

Elena-Bruyako commented on src/main/java/project/onlinebookstore/controller/BookController.java:
```diff
@@ -18,41 +22,71 @@
+    public List<BookDto> searchBooks(
+            @Parameter(description = "Represents an object whose fields represent"
+                    + " an array of search parameters")
+            BookSearchParameters searchParameters) {
```
```suggestion
            BookSearchParameters searchParameters, @ParameterObject @PageableDefault Pageable pageable) {
```

Elena-Bruyako commented on src/main/java/project/onlinebookstore/controller/BookController.java:
```diff
@@ -18,41 +22,71 @@
+            @Parameter(description = "Represents an object whose fields represent"
+                    + " an array of search parameters")
+            BookSearchParameters searchParameters) {
         return bookService.searchBooks(searchParameters);
```
```suggestion
        return bookService.searchBooks(searchParameters, pageable);
```

Elena-Bruyako commented on src/main/java/project/onlinebookstore/controller/BookController.java:
```diff
@@ -18,41 +22,71 @@
+@Tag(name = "Book Management",
+        description = "Contains book entity management operations")
 @RestController
 @RequestMapping("/books")
```
```suggestion
@RequestMapping("/books")
@Validated
```

Elena-Bruyako reviewed (changes requested):
Good job! Let's improve it

petro2061 committed:
added new annotations

petro2061 committed:
added new parameter into searchBooks method

petro2061 committed:
fixed problem with codestyle

petro2061 committed:
changed @NotBlank to @NotNull for BigDecimal field

petro2061 committed:
fixed problem with codestyle

petro2061 committed:
fix marge conflict

petro2061 commented:
Fixed problem with codestyle and added new annotation to BookController class. In CreateBookRequestDto class changed annotation NotBlant to NotNull becouse i recived error: "No validator could be found for constraint 'jakarta.validation.constraints.NotBlank' validating type 'java.math.BigDecimal'. Check configuration for 'price'"

Elena-Bruyako reviewed (changes requested):
see the comment

petro2061 committed:
fixed method updateBookById

petro2061 commented:
Added annotation @Positive to field Long id.

Elena-Bruyako reviewed (approved):
Good job
